.github/actions/maven-build/action.yml

```diff
@@ -0,0 +1,280 @@
+
+    - name: Remove running docker containers
+      if: inputs.reuse-testcontainers == 'true'
+      shell: bash
+      run: docker rm -f $(docker ps -a -q)
+      continue-on-error: true
+
+    - name: Echo Longest Tests run
+      if: env.SKIP_TESTS != 'true'
+      shell: bash
+      run: find . -name TEST-*.xml -exec grep -h testcase {} \; | awk -F '"' '{printf("%s#%s() - %.3fms\n", $4, $2, $6); }' | sort -n -k 3 | tail -20
+
+    - name: Set jacoco report name
+      if: inputs.jacoco-report-name != ''
+      id: get-jacoco-report-path
+      shell: bash
+      run: |
+        if [ "${{ inputs.jacoco-report-name }}" != "" ]
+        then
+          echo "path=${{ github.workspace }}/target/${{ inputs.jacoco-report-name }}.exec" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Merge Jacoco Reports
+      if: inputs.jacoco-report-name != ''
+      shell: bash
+      env:
+        MAVEN_RESOLVER_TRANSPORT_OPTS: ${{ inputs.maven-resolver-transport-options }}
+      run: |
+        mvn -N jacoco:merge -Djacoco.destFile=${{ steps.get-jacoco-report-path.outputs.path }} ${MAVEN_RESOLVER_TRANSPORT_OPTS}
+
+    - name: Update Jacoco Reports
+      if: inputs.jacoco-report-name != ''
+      shell: bash
+      env:
+        MAVEN_RESOLVER_TRANSPORT_OPTS: ${{ inputs.maven-resolver-transport-options }}
+      run: |
+        mvn jacoco:report -Djacoco.dataFile=${{ steps.get-jacoco-report-path.outputs.path }} \
+          -Djacoco-report-dir-suffix=${{ inputs.jacoco-report-name }} ${{ inputs.jacoco-report-options }} ${MAVEN_RESOLVER_TRANSPORT_OPTS}
+
+    - name: Upload target folders
+      if: inputs.target-folder-upload-name != ''
+      uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+      with:
+        name: ${{ inputs.target-folder-upload-name }}
+        path: |
+          ${{ github.workspace }}/**/target/classes
+          ${{ github.workspace }}/**/target/test-classes
+          ${{ github.workspace }}/**/target/site
+        retention-days: ${{ inputs.uploaded-artifacts-retention-days }}
+
+    - name: Get m2 inclusion pattern
+      id: m2-inclusion-pattern
+      shell: bash
+      env:
+        VERSION: ${{ steps.project-info.outputs.version }}
+      run: |
+        M2_GROUP_RELATIVE_PATH="$(yq e '.project.groupId | sub("\.", "/")' pom.xml)"
+        M2_GROUP_FULL_PATH="~/.m2/repository/${M2_GROUP_RELATIVE_PATH}"
+        INCLUSION_PATTERN="${M2_GROUP_FULL_PATH}/**/${VERSION}"
+        echo "group-path=$M2_GROUP_FULL_PATH" >> $GITHUB_OUTPUT
+        echo "pattern=$INCLUSION_PATTERN" >> $GITHUB_OUTPUT
+
+    - name: Upload M2 Build artifacts
+      uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+      if: inputs.m2-current-build-upload-name!= ''
+      with:
+        name: ${{ inputs.m2-current-build-upload-name}}
+        path: ${{ steps.m2-inclusion-pattern.outputs.pattern }}
+        retention-days: ${{ inputs.uploaded-artifacts-retention-days }}
```

Comment:
Not uploading jars files generated by the current build results in sonar goal slowing down because it attempts to download artifacts from remote.